src/plugins/workspace/public/utils.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { AppCategory } from '../../../core/public';
+
+/**
+ * Checks if a given feature matches the provided feature configuration.
+ *
+ * Rules:
+ * 1. `*` matches any feature.
+ * 2. Config starts with `@` matches category, for example, @management matches any feature of `management` category,
+ * 3. To match a specific feature, use the feature id, such as `discover`,
+ * 4. To exclude a feature or category, prepend with `!`, e.g., `!discover` or `!@management`.
+ * 5. The order of featureConfig array matters. From left to right, later configs override the previous ones.
+ *    For example, ['!@management', '*'] matches any feature because '*' overrides the previous setting: '!@management'.
+ */
+export const featureMatchesConfig = (featureConfigs: string[]) => ({
+  id,
+  category,
+}: {
+  id: string;
+  category?: AppCategory;
+}) => {
+  let matched = false;
+
+  for (const featureConfig of featureConfigs) {
+    // '*' matches any feature
+    if (featureConfig === '*') {
+      matched = true;
```

Comment:
Is it possible to return immediately and exit the loop, so we don't need to further iteration? the same as other if condition.  for example, remove the matched, just `return true;` here.  and `return false; ` for "No match found"

Reply:
@Flyingliuhub This is by design, see the comment: `5. The order of featureConfig array matters. From left to right, later configs override the previous ones.`

Reply:
@Flyingliuhub The function is a little bit confused, but the logic is correct. We can't `return true` or `return false` immediately. For example, if our rule is `['*', '!dev_tools']`, it means will match all the features except `dev_tools` feature. The loop is from left to right, if we `return true` when match `*`, the `dev_tools` feature will always be matched. The reason is the same in below if statements. We can't return directly, it's depends on the rest feature config.